src/main/java/nextmainfocus/jwt/JwtService.java

```diff
@@ -0,0 +1,115 @@
+package nextmainfocus.jwt;
+
+import java.security.Key;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.MalformedJwtException;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.UnsupportedJwtException;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import io.jsonwebtoken.security.SignatureException;
+
+@Service
+// @ConfigurationProperties("jwt")
+public class JwtService {
+	private static final Logger logger = LoggerFactory.getLogger(JwtService.class);
+
+	private long expiration = 18000000; // 5 hours in milliseconds 5 * 60 * 60 * 1000
+	private String secret = "3778217A25432A462D4A614E645267556B58703273357638792F423F4428472B";
+
+	public String authorizationHeader = "Authorization";
```

Comment:
all such vars could be final (const)

Reply:
Agree. All such vars should be extracted into an external config file (according to #25)
Today it is "Authorization," tomorrow that might be "authoriSation" 😋